requirements/requirements_git.txt

```
@@ -2,5 +2,5 @@ git+https://github.com/ansible/system-certifi.git@devel#egg=certifi
 # Remove pbr from requirements.in when moving ansible-runner to requirements.in
 git+https://github.com/ansible/ansible-runner.git@devel#egg=ansible-runner
 django-ansible-base @ git+https://github.com/ansible/django-ansible-base@devel#egg=django-ansible-base[rest-filters,jwt_consumer,resource-registry,rbac,feature-flags]
-awx-plugins-core @ git+https://github.com/ansible/awx-plugins.git@devel#egg=awx-plugins-core
+awx-plugins-core @ git+https://github.com/ansible/awx-plugins.git@devel#egg=awx-plugins-core[credentials-github-app]
```

This is a bit of a Frankenstein syntax. It's better to use the normal dep+extras spec. The `#egg` bit is a part of older syntax that existed pre `pkg @ url`, almost as a hack. The ecosystem doesn't even use the concept of eggs anymore except for when the legacy abstractions leak into the UI...

Here you go:
```suggestion
awx-plugins-core [credentials-github-app] @ git+https://github.com/ansible/awx-plugins.git
```

---

That argument would seem to apply equally to django-ansible-base specification on the line before. Do you agree?

---

Absolutely! I just didn't want to enlarge the PR scope.

---

How would you specify a custom branch? Like:

```
django-ansible-base [rest-filters,jwt_consumer,resource-registry,rbac,feature-flags] @ git+https://github.com/alancoding/django-ansible-base.git@feature_branch_for_testing
```

(don't know if this works)

---

I made an attempt at this, and it turns out the main issue is that our `updater.sh` script won't correctly comment out the git requirements in `requirements.txt`, so more work than I anticipated. But no objections.

---

Right.. Makes sense. I've been meaning to look into making the updater less complex.